tka/aum.go

```diff
@@ -161,14 +183,14 @@ func (a *AUM) StaticValidate() error {
 		if a.Key == nil {
 			return errors.New("AddKey AUMs must contain a key")
 		}
-		if a.KeyID != nil || a.State != nil || a.Votes != nil || a.Meta != nil {
+		if a.KeyID != nil || a.State != nil || a.Votes != nil || a.Meta != nil || a.NodeKey != nil {
```

Comment:
This is getting cumbersome, I wonder if we want to change the `AUM` struct to use some other approach to varying the fields for each `AUMKind`

I almost wonder if something like this would be good:
```golang
type AUM struct {
  MessageKind AUMKind
  PrevAUMHash []byte
  MessageParams []byte // or is there a cbor.RawMessage?
}
```

And then `MessageParams` would be one of several possible struct types depending on what `MessageKind` says.

Reply:
I originally opted not to nest message subtypes because that form usually lends itself to accidental type confusion issues. Additionally, I'm hesitant to change the layout of the AUM struct now as it would represent a massive refactor.

If the issue is mostly how cumbersome the checks are, what if we used reflection to invert it? So instead of listing the fields that arent allowed, we list the fields that are?

```go
if !a.OnlySet("NodeKey") {
  return errors.New("only NodeKey may be set on AddDenylistNodeKey AUMs")
}
```

wdyt?